src/HDFFile.cpp

```diff
@@ -280,6 +280,10 @@ void HDFFile::writeArrayOfAttributes(hdf5::node::Node const &Node,
           DType = AttrType.inner();
           writeAttrOfSpecifiedType(DType, Node, Name, StringSize, Encoding,
                                    Values);
+        } else if (auto AttrType = find<std::string>("dtype", Attribute)) {
```

Comment:
Pretty much a duplicate of the code above. I would look at unifying both "finds" out into a separate method or something similar.